### PR TITLE
week6 assignment

### PR DESCRIPTION
**Pull request by JafariM**

### CODE REVIEW

File: 03-task-manager/starter/controllers/tasks.js [hunk 1]
```diff
@@ -1,25 +1,51 @@
...
+const getTasks = asyncWrapper (async (req,res)=>{
+    const tasks = await Task.find({})
+    res.status(201).json({tasks})
+})
 
-const getSingleTask = (req,res)=>{
-    res.send('Get single task')
-}
+//get a task
+const getSingleTask = asyncWrapper(async(req,res)=>{
```
Comment (olearyma) [inline 03-task-manager/starter/controllers/tasks.js]: Missing next as parameter
```suggestion
const getSingleTask = asyncWrapper(async(req,res,next)=>{
```

File: 03-task-manager/starter/controllers/tasks.js [hunk 2]
```diff
@@ -1,25 +1,51 @@
...
-    res.status(201).json({task})
-}
+    res.status(201).json({task})     
+})
 
-const updateTask = (req,res)=>{
-    res.send('update task')
-}
+//update a task
+const updateTask = asyncWrapper(async(req,res)=>{
```
Comment (olearyma) [inline 03-task-manager/starter/controllers/tasks.js]: Missing next as parameter
```suggestion
const getSingleTask = asyncWrapper(async(req,res,next)=>{
```

File: 03-task-manager/starter/controllers/tasks.js [hunk 3]
```diff
@@ -1,25 +1,51 @@
...
+        return next(createCustomError(`No task with id : ${taskID}`, 404))
+    }
+    res.status(200).json({task})
+})
 
-const deleteTask= (req,res)=>{
-    res.send('delete task')
-}
+//remove a task
+const deleteTask= asyncWrapper(async(req,res)=>{
```
Comment (olearyma) [inline 03-task-manager/starter/controllers/tasks.js]: Missing next as parameter
```suggestion
const getSingleTask = asyncWrapper(async(req,res,next)=>{
```

**Review by olearyma — COMMENTED** [review]
Nice job on completing the task manager api

I tested the 404 handling and did find an issue.  
next was not defined as a parameter so the next call resulted in this error
"next is not defined"

The way I found this out was by added a console.log(err) statement inside errorHandlerMiddleware :

```
const errorHandlerMiddleware = (err, req, res, next) => {
  console.log(err);
  if (err instanceof CustomAPIError) {
    return res.status(err.statusCode).json({ msg: err.message })
  }
  return res.status(500).json({ msg: 'Something went wrong, please try again' })
}
```

After you update your code you can test using the following 
/api/v1/tasks/57636a9be7771a13933a641a

**Comment by JafariM** [comment]
Thank you, now I know what was wrong with the code

**Pull request merged**